cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -611,9 +626,12 @@ data TxOutValue era where
 
 deriving instance Eq   (TxOutValue era)
 deriving instance Show (TxOutValue era)
-deriving instance ToJSON (TxOutValue era)
 deriving instance Generic (TxOutValue era)
 
+instance ToJSON (TxOutValue era) where
+  toJSON (TxOutAdaOnly _ (Lovelace int)) = Aeson.Number $ fromInteger int
+  toJSON (TxOutValue _ val) = toJSON val
```

Comment:
:+1: yes this seems fine. There is already a sensible JSON instance for Value.

Though note that you don't need to re-implement the ToJSON instance for Lovelace as it already has one, just reuse it.